sw/host/tests/manuf/provisioning/src/main.rs

```diff
@@ -59,20 +72,71 @@ fn provisioning(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
     }
     device_pk_sec1_bytes.push(0x04); // This indicates the EC public key is not compressed.
     device_pk_sec1_bytes.reverse();
-    let _device_pk = PublicKey::<NistP256>::from_sec1_bytes(&device_pk_sec1_bytes)?;
+    let device_pk = PublicKey::<NistP256>::from_sec1_bytes(&device_pk_sec1_bytes)?;
+
+    // Peform ECDH to generate shared secret (AES) key.
+    let ecdh_shared_secret = diffie_hellman(hsm_sk.to_nonzero_scalar(), device_pk.as_affine());
+    let mut aes_key_bytes = Vec::from(ecdh_shared_secret.raw_secret_bytes().as_slice());
+    aes_key_bytes.reverse();
+    let aes_key = GenericArray::from_slice(aes_key_bytes.as_slice());
+
+    // Load encrypted RMA unlock token into a GenericArray.
+    let mut ciphertext = Vec::new();
+    for ciphertext_word in export_data.wrapped_rma_unlock_token.data {
+        ciphertext.extend(&ciphertext_word.to_le_bytes());
+    }
+    let plaintext = GenericArray::from_mut_slice(ciphertext.as_mut_slice());
+
+    // Decrypt the RMA unlock token, and convert it to a fixed array of 32-bit words.
+    let aes_ecb_cipher = Aes256Dec::new(aes_key);
+    aes_ecb_cipher.decrypt_block(plaintext);
+    let mut rma_unlock_token = [0u32; 4];
+    for (i, word) in rma_unlock_token.iter_mut().enumerate() {
+        *word = u32::from_le_bytes(plaintext[(i * 4)..((i * 4) + 4)].try_into().unwrap());
+    }
 
-    // TODO(#17393): Peform ECDH to generate shared secrete (AES) key.
-    // TODO(#17393): Decrypt RMA unlock token.
-    // TODO(#17393): Try to issue an LC transition to RMA to verify unlock token.
+    // Connect to JTAG LC TAP.
+    let jtag = transport.jtag(&opts.jtag_params)?;
+    jtag.connect(JtagTap::LcTap)?;
+
+    // Check the current LC state is Dev.
+    // We must wait for the lc_ctrl to initialize before the LC state is exposed.
+    wait_for_status(&jtag, Duration::from_secs(3), LcCtrlStatus::INITIALIZED)?;
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?,
+        DifLcCtrlState::Dev.redundant_encoding(),
```

Comment:
nit: Check if in `{DEV, PROD, PROD_END}`. We'll update the device side later.

Reply:
ah good catch, fixed.